synapse/storage/relations.py

```diff
@@ -312,6 +314,59 @@ def _get_aggregation_groups_for_event_txn(txn):
             "get_aggregation_groups_for_event", _get_aggregation_groups_for_event_txn
         )
 
+    @cachedInlineCallbacks(tree=True)
```

Comment:
Oh, I mainly just did it because I needed it for the other ones in this class (so that we can use `invalidate_many`). We could remove it, but means adding `sender` to the event stream replication, which is probably fine and useful thing to do.

In fact, why don't I just pull the type and sender from the `events` table and then we don't need to pass them in as parameters.

> I think the cache is going to be keyed on all three params - is that correct?

Yup